### PR TITLE
[cli] - Don't escape special characters when printing JSON

### PR DESCRIPTION
**Pull request by komalali**
# Description

Fixes https://github.com/pulumi/pulumi/issues/5474

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by EvanBoyle** [comment]
Huh, I wonder if the failures seen in https://github.com/pulumi/pulumi/issues/6768 are at all related to this category of bug occurring elsewhere in the engine event code.

I do see unicode in the error trace posted by one of the users (`\u003c`) in that thread.

**Comment by komalali** [comment]
>Huh, I wonder if the failures seen in #6768 are at all related to this category of bug occurring elsewhere in the engine event code.

The escaped unicode is certainly present in those engine events and likely should be corrected. I don't believe that's the source of that error though. Every single diagnostic event contains escaped unicode that you referenced, but they rarely result in an error. And I haven't seen any evidence that the break in json resulting in the error is at the location of the escaped unicode.

File: pkg/cmd/pulumi/util.go [hunk 1]
```diff
@@ -775,13 +775,25 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
...
+	encoder.SetIndent("", "  ")
+	err := encoder.Encode(v)
+	if err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
 // printJSON simply prints out some object, formatted as JSON, using standard indentation.
 func printJSON(v interface{}) error {
```
Comment (t0yv0) [inline pkg/cmd/pulumi/util.go:791]: Nit: perhaps put "indent" in the name of the func to further emphasize what it does.

**Review by t0yv0 — APPROVED** [review]
LGTM

File: pkg/cmd/pulumi/util.go [hunk 2]
```diff
@@ -775,13 +775,25 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
 	return c
 }
 
+func makeJSONString(v interface{}) (string, error) {
+	out := bytes.NewBuffer([]byte{})
+	encoder := json.NewEncoder(out)
```
Comment (pgavlin) [inline pkg/cmd/pulumi/util.go:780]: The zero value for `bytes.Buffer` is a usable buffer:

```suggestion
	var out bytes.Buffer
	encoder := json.NewEncoder(&out)
```

File: pkg/cmd/pulumi/util.go [hunk 3]
```diff
@@ -775,13 +775,25 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
...
+	encoder.SetEscapeHTML(false)
+	encoder.SetIndent("", "  ")
+	err := encoder.Encode(v)
+	if err != nil {
```
Comment (pgavlin) [inline pkg/cmd/pulumi/util.go:784]:
```suggestion
	if err := encoder.Encode(v); err != nil {
```

**Review by pgavlin — APPROVED** [review]
LGTM overall--two minor suggestions.

File: pkg/cmd/pulumi/util.go (hunk 1 above)
Comment (komalali) [reply pkg/cmd/pulumi/util.go:791]: Since this function already existed and is used in many places in the CLI I'd rather not change them all here :)

**Pull request merged**